### ISSUE TITLE
 Exporting the operator 'aten::_convolution_mode' to ONNX opset version 14 is not supported.

### ISSUE THREAD

**Issue by Omaressam2002** [pytorch/pytorch#108975]
### 🐛 Describe the bug

```python

import torch
model = darkNet()


torch.onnx.export(model,               # model being run
                  torch.randn(1, 3, 448, 448), # dummy input (required)
                  "DN.onnx",   # where to save the model (can be a file or file-like object)
                  export_params=True)
```

### Versions

.

**Comment by thiagocrepaldi** [comment]
@Omaressam2002 please provide a full working script, including the `darkNet` model to allow us to reproduce and investigate the issue

**Comment by Omaressam2002** [comment]
i fixed it!! the issue was with using padding = 'same' in the convolution layers , instead try and replace 'same' with an integer that yeilds the same output size. thanks for the reply tho :))

**Comment by thiagocrepaldi** [comment]
@Omaressam2002 a full repro would still be valuable as we could improve our code to be more clear on how to use

**Comment by Omaressam2002** [comment]
class darkNet(nn.Module): #Architectures
    def __init__(self,classes=1000,init_weight=False):
        super(darkNet, self).__init__()
        self.darkNet = nn.Sequential(
             ConvBlock(7,3,64,2,3),
             nn.MaxPool2d(kernel_size=2,stride=2),
             ConvBlock(3,64,192,padding=1),
             nn.MaxPool2d(kernel_size=2,stride=2),
             ConvBlock(1,192,128),
             ConvBlock(3,128,256,padding=1),
             ConvBlock(1,256,256),
             ConvBlock(3,256,512,padding=1),
             nn.MaxPool2d(kernel_size=2,stride=2),
             
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),

            ConvBlock(1,512,512),
            ConvBlock(3,512,1024,padding=1),
            nn.MaxPool2d(kernel_size=2,stride=2),

            ConvBlock(1,1024,512),
            ConvBlock(3,512,1024,padding=1),
            ConvBlock(1,1024,512),
            ConvBlock(3,512,1024,padding=1)
        )
        self.classifier = nn.Sequential(
            *self.darkNet,
            GlobalAvgPool2d(),
            nn.Linear(1024, classes)
        )
        
        
        
        if init_weight:
            self._initialize_weights()

    def forward(self, x):
        return self.classifier(x)
    
    def _initialize_weights(self):
        for m in self.modules():
            if isinstance(m, ConvBlock):
                nn.init.kaiming_normal_(m.conv.weight, mode='fan_in', nonlinearity='leaky_relu')
                nn.init.constant_(m.batchNorm.weight, 1)
                nn.init.constant_(m.batchNorm.bias, 0)
            elif isinstance(m, nn.Linear):
                nn.init.normal_(m.weight, 0, 0.01)
                nn.init.constant_(m.bias, 0)


class ConvBlock(nn.Module):
    def __init__(self,size,ch_in,ch_out,stride=1,padding=0):
        super(ConvBlock,self).__init__()
        self.conv = nn.Conv2d(ch_in,ch_out,size,stride,padding,bias=False) #batch norm so bias = false , half precision
        self.batchNorm = nn.BatchNorm2d(ch_out)
        self.lrelu = nn.LeakyReLU(0.1)

    def forward(self,x):
        return self.lrelu(self.batchNorm(self.conv(x)))

**Comment by Omaressam2002** [comment]
class ConvBlock(nn.Module):
    def __init__(self,size,ch_in,ch_out,stride=1,padding=0):
        super(ConvBlock,self).__init__()
        self.conv = nn.Conv2d(ch_in,ch_out,size,stride,padding,bias=False) #batch norm so bias = false , half precision
        self.batchNorm = nn.BatchNorm2d(ch_out)
        self.lrelu = nn.LeakyReLU(0.1)

    def forward(self,x):
        return self.lrelu(self.batchNorm(self.conv(x)))



class darkNet(nn.Module): #Architectures
    def __init__(self,classes=1000,init_weight=False):
        super(darkNet, self).__init__()
        self.darkNet = nn.Sequential(
             ConvBlock(7,3,64,2,3),
             nn.MaxPool2d(kernel_size=2,stride=2),
             ConvBlock(3,64,192,padding=1),
             nn.MaxPool2d(kernel_size=2,stride=2),
             ConvBlock(1,192,128),
             ConvBlock(3,128,256,padding=1),
             ConvBlock(1,256,256),
             ConvBlock(3,256,512,padding=1),
             nn.MaxPool2d(kernel_size=2,stride=2),
             
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),
            ConvBlock(1,512,256),
            ConvBlock(3,256,512,padding=1),

            ConvBlock(1,512,512),
            ConvBlock(3,512,1024,padding=1),
            nn.MaxPool2d(kernel_size=2,stride=2),

            ConvBlock(1,1024,512),
            ConvBlock(3,512,1024,padding=1),
            ConvBlock(1,1024,512),
            ConvBlock(3,512,1024,padding=1)
        )
        self.classifier = nn.Sequential(
            *self.darkNet,
            GlobalAvgPool2d(),
            nn.Linear(1024, classes)
        )
        
        
        
        if init_weight:
            self._initialize_weights()

    def forward(self, x):
        return self.classifier(x)
    
    def _initialize_weights(self):
        for m in self.modules():
            if isinstance(m, ConvBlock):
                nn.init.kaiming_normal_(m.conv.weight, mode='fan_in', nonlinearity='leaky_relu')
                nn.init.constant_(m.batchNorm.weight, 1)
                nn.init.constant_(m.batchNorm.bias, 0)
            elif isinstance(m, nn.Linear):
                nn.init.normal_(m.weight, 0, 0.01)
                nn.init.constant_(m.bias, 0)

**Comment by Omaressam2002** [comment]
here you go

**Comment by thiagocrepaldi** [comment]
Closing as there is a workaround

`torch.onnx.export` is in maintenance mode and we don't plan to add new operators/features or fix complex issues.

Please try the new ONNX exporter and reopen this issue with a full repro if it also doesn't work for you: [quick torch.onnx.dynamo_export API tutorial](https://pytorch.org/tutorials/beginner/onnx/export_simple_model_to_onnx_tutorial.html)

**Comment by jiaduob** [comment]
Thanks a lot. It works.
